community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/References.java

```diff
@@ -134,4 +140,79 @@ static boolean hasDirectFlag( long groupReference )
         assert groupReference != NO_ID;
         return (groupReference & DIRECT_FLAG) != 0L;
     }
+
+    /**
+     * Encode a node id as a property reference.
+     *
+     * This allows us to encode a node reference as a property reference so that we can do lookups of changes in the
+     * transaction state.
+     * @param reference the reference to encode
+     * @return The encoded reference
+     */
+    static long setNodeFlag( long reference )
+    {
+        return reference | NODE_MARKER | FLAG_MARKER;
+    }
+
+    /**
+     * Checks if the property reference really is a node reference
+     *
+     * @param nodeReference The reference to check
+     * @return <tt>true</tt>if the reference refers to a node otherwise <tt>false</tt>
+     */
+    static boolean hasNodeFlag( long nodeReference )
+    {
+        assert nodeReference != NO_ID;
+        return (nodeReference & NODE_MARKER) != 0L;
+    }
+
+    /**
+     * Encode a relationship id as a property reference.
+     *
+     * This allows us to encode a relationship reference as a property reference so that we can do lookups of changes in the
+     * transaction state.
+     * @param relationshipReference the reference to encode
+     * @return The encoded reference
+     */
+    static long setRelationshipFlag( long relationshipReference )
+    {
+        return relationshipReference | RELATIONSHIP_MARKER | FLAG_MARKER;
+    }
+
+    /**
+     * Checks if the property reference really is a relationship reference
+     *
+     * @param reference The reference to check
+     * @return <tt>true</tt>if the reference refers to a relationship otherwise <tt>false</tt>
+     */
+    static boolean hasRelationshipFlag( long reference )
```

Comment:
This `reference` parameter not updated yet.

Reply:
We are splitting hairs here but in this case we don't know yet if it is a relationship reference it is the point of the function to tell us that. So in this case and for `hasNodeFlag` I prefer just `reference`

Reply:
I'm happy now 🎉  I just thought you had forgotten to check them because some of them where changed and some not 😅